internal/manager/manager.go

```diff
@@ -0,0 +1,169 @@
+package manager
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/sourcegraph/conc/pool"
+
+	"github.com/deckhouse/d8-lint/internal/flags"
+	"github.com/deckhouse/d8-lint/internal/logger"
+	"github.com/deckhouse/d8-lint/internal/module"
+	"github.com/deckhouse/d8-lint/pkg/config"
+	"github.com/deckhouse/d8-lint/pkg/errors"
+	"github.com/deckhouse/d8-lint/pkg/linters/container"
+	"github.com/deckhouse/d8-lint/pkg/linters/copyright"
+	"github.com/deckhouse/d8-lint/pkg/linters/modules"
+	no_cyrillic "github.com/deckhouse/d8-lint/pkg/linters/no-cyrillic"
+	"github.com/deckhouse/d8-lint/pkg/linters/object"
+	"github.com/deckhouse/d8-lint/pkg/linters/openapi"
+	"github.com/deckhouse/d8-lint/pkg/linters/probes"
+	"github.com/deckhouse/d8-lint/pkg/linters/rbac"
+)
+
+const (
+	ChartConfigFilename = "Chart.yaml"
+	ModuleYamlFilename  = "module.yaml"
+	HooksDir            = "hooks"
+	ImagesDir           = "images"
+)
+
+type Manager struct {
+	cfg     *config.Config
+	Linters LinterList
+	Modules []*module.Module
+
+	lintersMap map[string]Linter
+}
+
+func NewManager(dirs []string, cfg *config.Config) *Manager {
+	m := &Manager{
+		cfg: cfg,
+	}
+
+	// fill all linters
+	m.Linters = []Linter{
+		openapi.New(&cfg.LintersSettings.OpenAPI),
+		no_cyrillic.New(&cfg.LintersSettings.NoCyrillic),
+		copyright.New(&cfg.LintersSettings.Copyright),
+		probes.New(&cfg.LintersSettings.Probes),
+		container.New(&cfg.LintersSettings.Container),
+		object.New(&cfg.LintersSettings.Object),
+		modules.New(&cfg.LintersSettings.Modules),
+		rbac.New(&cfg.LintersSettings.Rbac),
+	}
+
+	m.lintersMap = make(map[string]Linter)
+	for _, linter := range m.Linters {
+		m.lintersMap[strings.ToLower(linter.Name())] = linter
+	}
+
+	m.Linters = make(LinterList, 0)
+	for _, linter := range m.lintersMap {
+		m.Linters = append(m.Linters, linter)
+	}
+
+	var paths []string
+
+	for i := range dirs {
+		dir, err := homedir.Expand(dirs[i])
+		if err != nil {
+			logger.ErrorF("Failed to expand home dir: %v", err)
+			continue
+		}
+		result, err := getModulePaths(dir)
+		if err != nil {
+			logger.ErrorF("Error getting module paths: %v", err)
+		}
+		paths = append(paths, result...)
```

Comment:
if we have an error from get module path - we append nil. it's ok?

Reply:
added directory skipping